frontend/src/services/division/division.service.ts

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from 'react-query';
+import { OrderDirection } from '../../common/enums/order-direction.enum';
+import { DivisionType } from '../../components/Division';
+import { getDivisionData } from './division.api';
+
+export const useDivisionDataQuery = (
+  limit: number,
+  page: number,
+  orderBy: string,
+  orderDirection: OrderDirection,
+  divisionType: DivisionType,
+) => {
+  return useQuery(['divisions', limit, page, orderBy, orderDirection, divisionType], () =>
+    getDivisionData(limit, page, orderBy, orderDirection, divisionType),
```

Comment:
This should only be called if limit, page or divisionType are provided, the rest of them are optional.